nodebuilder/core/module.go

```diff
@@ -0,0 +1,56 @@
+package core
+
+import (
+	"context"
+
+	"go.uber.org/fx"
+
+	"github.com/celestiaorg/celestia-node/core"
+	"github.com/celestiaorg/celestia-node/header"
+	headercore "github.com/celestiaorg/celestia-node/header/core"
+	"github.com/celestiaorg/celestia-node/libs/fxutil"
+	"github.com/celestiaorg/celestia-node/nodebuilder/node"
+)
+
+// Module collects all the components and services related to managing the relationship with the Core node.
+func Module(tp node.Type, cfg *Config, options ...fx.Option) fx.Option {
+	// sanitize config values before constructing module
+	cfgErr := cfg.ValidateBasic()
+
+	baseComponents := fx.Options(
+		fx.Supply(*cfg),
+		fx.Error(cfgErr),
+		fx.Options(options...),
+	)
```

Comment:
It is not guaranteed when an Invoke occurs -- we want the application to return if an error is found (via providing it to fx.Error). This is why we do this via fx.Error instead of invoke. Also all constructors depend on a sanitised config and would not be able to build without it so we need that to happen _before_.

```
// Similar to invocations, errors are applied in order. All Provide and Invoke
// options registered before or after an Error option will not be applied.
```

Reply:
> It is not guaranteed when an Invoke occurs 

> // Similar to invocations, errors are applied in order.

?

Reply:
Resolved via slack?

Reply:
Yes